engine-mr/src/main/java/org/apache/kylin/engine/mr/streaming/RowRecordReader.java

```diff
@@ -82,51 +82,72 @@ public RowRecordReader(CubeDesc cubeDesc, Path path, FileSystem fileSystem) thro
 
-        List<DimensionMetaInfo> allDimensions = basicCuboidMetaInfo.getDimensionsInfo();
-        Map<String, DimensionEncoding> dimensionEncodingMap = getDimensionEncodings(fragmentMetaInfo, allDimensions,
-                dictInputStream);
-        dimensionColumnReaders = Lists.newArrayList();
-        dimensionColumnReaderItrs = Lists.newArrayList();
-        dimensionEncodings = Lists.newArrayList();
-        for (DimensionMetaInfo dimensionMetaInfo : allDimensions) {
-            FSDataInputStream dimInputStream = fs.open(dataFilePath);
-            String dimName = dimensionMetaInfo.getName();
-            DimensionEncoding dimEncoding = dimensionEncodingMap.get(dimName);
-            ColumnarStoreDimDesc dimDesc = new ColumnarStoreDimDesc(dimEncoding.getLengthOfEncoding(),
-                    dimensionMetaInfo.getCompressionType());
-            ColumnDataReader dimDataReader = dimDesc.getDimReaderFromFSInput(dimInputStream,
-                    dimensionMetaInfo.getStartOffset(), dimensionMetaInfo.getDataLength(),
-                    (int) basicCuboidMetaInfo.getNumberOfRows());
-            dimensionColumnReaders.add(dimDataReader);
-            dimensionColumnReaderItrs.add(dimDataReader.iterator());
-            dimensionEncodings.add(dimEncoding);
-        }
-        rowDimensionValues = new String[dimensionColumnReaders.size()];
+            List<DimensionMetaInfo> allDimensions = basicCuboidMetaInfo.getDimensionsInfo();
+            Map<String, DimensionEncoding> dimensionEncodingMap = getDimensionEncodings(fragmentMetaInfo, allDimensions,
+                    dictInputStream);
+            dimensionColumnReaders = Lists.newArrayList();
+            dimensionColumnReaderItrs = Lists.newArrayList();
+            dimensionEncodings = Lists.newArrayList();
+            for (DimensionMetaInfo dimensionMetaInfo : allDimensions) {
+                FSDataInputStream dimInputStream = null;
+                try {
+                    dimInputStream = fs.open(dataFilePath);
+                    String dimName = dimensionMetaInfo.getName();
+                    DimensionEncoding dimEncoding = dimensionEncodingMap.get(dimName);
+                    ColumnarStoreDimDesc dimDesc = new ColumnarStoreDimDesc(dimEncoding.getLengthOfEncoding(),
+                            dimensionMetaInfo.getCompressionType());
+                    ColumnDataReader dimDataReader = dimDesc.getDimReaderFromFSInput(dimInputStream,
+                            dimensionMetaInfo.getStartOffset(), dimensionMetaInfo.getDataLength(),
+                            (int) basicCuboidMetaInfo.getNumberOfRows());
+                    dimensionColumnReaders.add(dimDataReader);
+                    dimensionColumnReaderItrs.add(dimDataReader.iterator());
+                    dimensionEncodings.add(dimEncoding);
+                } finally {
+                    if (null != dimInputStream) {
+                        dimInputStream.close();
```

Comment:
Should NOT close `dimInputStream ` here, otherwise `dimensionColumnReaderItrs` will be broken.